ts/packages/anchor/src/provider.ts

```diff
@@ -1,50 +1,74 @@
 import {
-  Connection,
-  Signer,
-  PublicKey,
+  Base58EncodedAddress,
+  createDefaultRpcTransport,
+  createSolanaRpc,
+  setTransactionFeePayer,
+  setTransactionLifetimeUsingBlockhash,
+  signTransaction,
   Transaction,
-  TransactionSignature,
-  ConfirmOptions,
-  SimulatedTransactionResponse,
-  Commitment,
-  SendTransactionError,
-  SendOptions,
-  VersionedTransaction,
-  RpcResponseAndContext,
-} from "@solana/web3.js";
+} from "web3js-experimental";
 import { bs58 } from "./utils/bytes/index.js";
-import { isBrowser, isVersionedTransaction } from "./utils/common.js";
-import {
-  simulateTransaction,
-  SuccessfulTxSimulationResponse,
-} from "./utils/rpc.js";
+import { isBrowser } from "./utils/common.js";
+import { SuccessfulTxSimulationResponse } from "./utils/rpc.js";
+
+// Type helpers for non-exported Web3 JS types.
+// As of right now, these types will not be exported from the new API.
+type PromiseType<T extends Promise<unknown>> = T extends Promise<infer R>
```

Comment:
Can use `Awaited` helper type that's built into TypeScript here.

https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype